algebra/tests/matrix/tools.c

```diff
@@ -33,6 +36,48 @@ void algebraTests_valFill(matrix_t *M, float val)
 }
 
 
+int algebraTests_createAndFill(matrix_t *M, unsigned int rows, unsigned int cols, float *vals)
+{
+	if (matrix_bufAlloc(M, rows, cols) != BUF_ALLOC_OK) {
+		return EXIT_FAILURE;
+	}
+
+	algebraTests_valsFill(M, vals);
+
+	return EXIT_SUCCESS;
+}
+
+
+void algebraTests_valsFill(matrix_t *M, float *vals)
+{
+	int rowsNum, colsNum, row, col;
+	int i = 0;
+
+	rowsNum = matrix_rowsGet(M);
+	colsNum = matrix_colsGet(M);
+
+	for (row = 0; row < rowsNum; row++) {
+		for (col = 0; col < colsNum; col++) {
+			*matrix_at(M, row, col) = vals[i];
+			i++;
+		}
+	}
+}
```

Comment:
You can merge it with `algebraTests_valFill()` as follows:
```suggestion
void algebraTests_buffFill(matrix_t *M, float *vals, unsigned int n)
{
	int rowsNum, colsNum, row, col;
	int i = 0;

	rowsNum = matrix_rowsGet(M);
	colsNum = matrix_colsGet(M);

	for (row = 0; row < rowsNum; row++) {
		for (col = 0; col < colsNum; col++) {
			*matrix_at(M, row, col) = (n > 1) ? vals[i++] : vals[i];
		}
	}
}
```
This way less code is reused, as there is very little difference between those functions.